### PR TITLE
Prettier updates

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This PR makes it so that you can't push to DCR until your code passes `prettier` validation.

It also runs prettier from the CLI to fix all the existing non prettiness.

## Why does this happen?
Because 1) not everyone is a VSCode user and 2) even if you are, the checks can often get stuck and stop formatting on save. Rather than punish people here, hopefully the checks show them useful messages to help them progress. In addition, I've added a `yarn prettier:fix` command in yarn and updated `make fix` as well to make dealing with these warning easier and also updated the README to help highlight what to do.

## Why do we care?
The problem with allowing code on `main` that isn't pretty is that these changes will get picked up in unrelated PRs, cluttering things and making it hard to review.

### CODE REVIEW

**Review by shtukas — APPROVED** [review]

**Comment by oliverlloyd** [comment]
[Must be pretty](https://trello.com/c/0ZWpSIAr/2234-must-be-pretty)

**Review by liywjl — APPROVED** [review]
I like pretty things 👍

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# Include all folders
+!/**/*/
+
+# But exclude node_modules folders anywhere
+node_modules/
+
+# Include specific file extensions if they appear under a "src" folder
+!/src/**/*.ts
+!/src/**/*.tsx
+!/src/**/*.js
```
Comment (gtrufitt) [inline .prettierignore:13]: When you include all folders, do you not get these included by default?

**Review by gtrufitt — APPROVED** [review]

File: .prettierignore (hunk 1 above)
Comment (oliverlloyd) [reply .prettierignore:13]: it's a little counter intuitive for sure but no, you need _both_ statements (include all folders and don't exclude these files) to get the result we want 🤷

Comment (oliverlloyd) [reply .prettierignore:13]: I've added a comment to `.prettierignore` with a link to a Github discussion talking about how this file works

**Pull request merged**